### PR TITLE
3344 task migrate simulation to postgres

### LINKED ISSUE #3344: [TASK]: Migrate `Simulation` to postgres
**Describe the task**
Move `Simulation`s from ES to postgres in a simlar fashion as workflows were done in #3261

### PR DESCRIPTION
**Pull request by dvince2**
# Description

* Moving simulations over to postgres

There is a minor bit of cleanup to go, but, I'd love a sanity check while I do that :)

Resolves #3344

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/workflow/Workflow.java [hunk 1]
```diff
@@ -46,12 +42,22 @@ public class Workflow extends TerariumAsset {
 	@JdbcTypeCode(SqlTypes.JSON)
 	private List<WorkflowEdge> edges;
 
+	@OneToOne(mappedBy = "workflow")
+	@Where(clause = "deleted_on IS NULL")
+	@ToString.Exclude
+	@JsonManagedReference
+	private Simulation simulation;
```
Comment (mwdchang) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/workflow/Workflow.java:49]: I think there are some miscommunication here, a workflow should not have a direct reference to a Simulation. 

The reverse is also true that a Simulation does not need a direct reference to a Workflow.

**Review by mwdchang — CHANGES_REQUESTED** [review]

File: packages/client/hmi-client/src/types/Types.ts [hunk 2]
```diff
@@ -509,12 +507,9 @@ export interface Simulation {
     startTime?: Date;
     completedTime?: Date;
     engine: SimulationEngine;
-    workflowId: string;
     userId?: string;
     projectId?: string;
-    createdOn?: Date;
-    updatedOn?: Date;
-    deletedOn?: Date;
+    workflow: Workflow;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/types/Types.ts:512]: Why are we saving a full workflow within the `Simulation`?

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java [hunk 3]
```diff
@@ -72,31 +68,38 @@ public class Simulation implements Serializable {
 	@Enumerated(EnumType.STRING)
 	private SimulationEngine engine;
 
-	@JsonAlias("workflow_id")
-	private UUID workflowId;
+	@OneToOne
+	@JoinColumn(name = "workflow_id", nullable = true)
+	@JsonBackReference
+	private Workflow workflow;
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java:74]: This come back to my other comment, why the full workflow is saved here?

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java [hunk 4]
```diff
@@ -72,31 +68,38 @@ public class Simulation implements Serializable {
...
 
 	@JsonAlias("user_id")
 	@TSOptional
 	private String userId;
 
 	@JsonAlias("project_id")
 	@TSOptional
-	private UUID projectId;
+	private UUID projectId; // TODO this can probably be joined to the project table soon?
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java:82]: Do you mean via `ProjectAsset`?

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServiceWithSearch.java [hunk 5]
```diff
@@ -37,10 +38,11 @@ public TerariumAssetServiceWithSearch(
 			final ElasticsearchConfiguration elasticConfig,
 			final ElasticsearchService elasticService,
 			final ProjectAssetService projectAssetService,
+			final S3ClientService s3ClientService,
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServiceWithSearch.java:41]: Why do we need the `S3ClientService` now?

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServiceWithoutSearch.java
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServiceWithoutSearch.java:1]: Where does all this S3 manipulation coming from. I thought it was setup somewhere before, but I cannot found from where you move this here.

**Review by YohannParis — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java (hunk 4 above)
Comment (dvince2) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java:78]: `ProjectAsset` may be able to go away... they were stubs holding these things in postgres!

File: packages/server/src/test/java/software/uncharted/terarium/hmiserver/service/data/WorkflowServiceTests.java [hunk 6]
```diff
@@ -177,9 +179,10 @@ public void testItCanDeleteWorkflow() throws Exception {
 	public void testItCanCloneWorkflow() throws Exception {
 
 		Workflow workflow = createWorkflow();
-
 		workflow = workflowService.createAsset(workflow);
 
+		workflowService.updateAsset(workflow);
```
[the remark this answers is not in the document]
Comment (dvince2) [reply packages/server/src/test/java/software/uncharted/terarium/hmiserver/service/data/WorkflowServiceTests.java]: Removing!
When i though there needed to be a link between simulation and workflow there was some simulation magic happening here.

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java [hunk 7]
```diff
@@ -81,22 +75,27 @@ public class Simulation implements Serializable {
...
-	@Schema(accessMode = Schema.AccessMode.READ_ONLY)
-	@Column(columnDefinition = "TIMESTAMP WITH TIME ZONE")
-	private Timestamp updatedOn;
+		cloneSuperFields(clone);
 
-	@TSOptional
-	@Schema(accessMode = Schema.AccessMode.READ_ONLY)
-	@Column(columnDefinition = "TIMESTAMP WITH TIME ZONE")
-	private Timestamp deletedOn;
+		clone.setDescription(this.description); // done
```
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java:86]: Should we remove the // done comments?

Comment (dvince2) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java:86]: oh, definitely.

**Review by kbirk — APPROVED** [review]

**Pull request merged**